File: Reference-Android-Project-Java/app/src/main/java/xyz/fay/reference/utils/AssetProvider.java
package xyz.fay.reference.utils;

import android.content.Context;
import android.content.res.AssetManager;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;

public class AssetProvider {
    private final static String MOCK = "mock";

    @Nullable
    public static String loadFile(@NonNull Context context, @NonNull String fileName) {
        try {
            AssetManager assetManager = context.getAssets();
            InputStream inputStream = assetManager.open(fileName);
            InputStreamReader inputStreamReader = new InputStreamReader(inputStream);
            BufferedReader bufferedReader = new BufferedReader(inputStreamReader);
            StringBuilder stringBuilder = new StringBuilder();
            String line;
            while ((line = bufferedReader.readLine()) != null) {
                stringBuilder.append(line);
            }
            bufferedReader.close();
            inputStreamReader.close();
            inputStream.close();
            return stringBuilder.toString();
        } catch (IOException e) {
            e.printStackTrace();
        }
        return null;
    }

    @Nullable
    public static String loadJsonFile(@NonNull Context context, @NonNull String fileName) {
        return loadFile(context, MOCK + "/" + fileName);
    }
}

File: Reference-Android-Kotlin/app/src/main/java/xyz/fay/reference/utils/AssetProvider.kt
package xyz.fay.reference.utils

import android.content.Context
import java.io.BufferedReader
import java.io.InputStreamReader

class AssetProvider {
    companion object {
        private const val mock = "mock"

        fun loadFile(context: Context, fileName: String): String {
            val manager = context.assets
            val inputStream = manager.open(fileName)
            val inputStreamReader = InputStreamReader(inputStream)
            val bufferedReader = BufferedReader(inputStreamReader)
            val data = bufferedReader.use { it.readText() }
            bufferedReader.close()
            inputStreamReader.close()
            inputStream.close()
            return data
        }

        fun loadJsonFile(context: Context, fileName: String) =
            loadFile(context, "$mock/$fileName")
//            loadFile(context, "${mock}/${fileName}")
    }
}
